### PR TITLE
preserve dates for mean and median

### LINKED ISSUE #1112: The mean and median reducers convert dates to numbers
These should probably produce dates if the input is dates (like min and max do).

### PR DESCRIPTION
**Pull request by mbostock**
Fixes #1112.

### CODE REVIEW

File: src/transforms/group.js [hunk 1]
```diff
@@ -319,6 +320,15 @@ function reduceAccessor(f) {
...
 }
 
+function reduceMaybeTemporalAccessor(f) {
+  return {
+    reduce(I, X) {
+      const x = f(I, (i) => X[i]);
+      return isTemporal(X) ? new Date(x) : x;
+    }
+  };
+}
```
Comment (Fil) [inline src/transforms/group.js:330]: small optimization:
```suggestion
function reduceMaybeTemporalAccessor(f) {
  return {
    scope: "data",
    reduce(I, X, context) {
      if (arguments.length === 2) return isTemporal(X);
      const x = f(I, (i) => X[i]);
      return context ? new Date(x) : x;
    }
  };
}
```

if we do this, we'll also have to initialize the context in https://github.com/observablehq/plot/blob/21dc1c70f4de1ec2a63ae49fce6f04089bed3d5f/src/channel.js#L66

Comment (mbostock) [reply src/transforms/group.js:330]: I’d prefer to keep this simple for now.

**Review by Fil — APPROVED** [review]
Needs a test plot, but I believe that will be with one of the maps (#1111)?

**Comment by mbostock** [comment]
I added a few unit tests.

**Pull request merged**